src/libsystemd/sd-netlink/generic-netlink.c

```diff
@@ -0,0 +1,104 @@
+#include <linux/genetlink.h>
+
+#include "sd-netlink.h"
+#include "netlink-internal.h"
+#include "alloc-util.h"
+
+typedef struct {
+        const char* name;
+        uint8_t version;
+} genl_family;
+
+static const genl_family genl_families[] = {
+        [SD_GENL_ID_CTRL] = { .name = "", .version = 1 },
+        [SD_GENL_WIREGUARD] = { .name = "wireguard", .version = 1 },
+};
+
+int sd_genl_socket_open(sd_netlink **ret) {
```

Comment:
hmm, any chance you can split out the genl stuff into a separate commit in this PR?

Reply:
we have genl PR already https://github.com/systemd/systemd/pull/5593

Reply:
That genl implementation seems to be missing some things that this one has. I'd recommend merging this first, and then rebasing the ipoverfoo stuff ontop of this.